common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java

```diff
@@ -95,6 +105,28 @@ public static Map<String, Object> fromJson(String jsonStr, String defaultKey) {
         return result;
     }
 
+    public static Map<String, String> filteredParameterMap(Map<String, ?> parameterObjs, Set<String> allowedList) {
+        Map<String, String> parameters = new HashMap<>();
+        Set<String> filteredKeys = new HashSet<>(parameterObjs.keySet());
+        filteredKeys.retainAll(allowedList);
+        for (String key : filteredKeys) {
```

Comment:
Seems no need to add a new HashSet `filteredKeys`, just need to iterate the `allowedList` 
```
    for (String key : allowedList) {
        if (parameterObjs.containsKey(key)) {
```

Reply:
`filteredKeys` is the intersection of Map keys and allowedList, it will be shorter.